ase/sw/ase_common.h

```diff
@@ -404,7 +392,7 @@ extern uint64_t *umsg_umas_vbase;
 // ASE buffer valid/invalid indicator
 // When a buffer is 'allocated' successfully, it will be valid, when
 // it is deallocated, it will become invalid.
-#define ASE_BUFFER_VALID        0xFFFF
+#define ASE_BUFFER_VALID        0x1
```

Comment:
Enum?

Reply:
Is this a magic sequence to detect unitialized memory (in which case it should have values that look a bit more magic)? Or is it just a boolean?

Reply:
Boolean is enough for this case. I changed it to be 0x01 because wsmeta_t's valid field use a '0' and '1'. So I didn't want to break any code in case it is used in some special way.

Reply:
I suggest leave this as 32-bit number for 2 reasons:
1. This has to be packed into a struct and communicated over a named pipe, so things need to be predictable.
2. Future expansion of stateful data as needed.

Reply:
In struct buffer-t definition, "int32_t valid;' is used for buffer->valid bit. No matter what value you assign it to this field, it will use a 32 bit data for it.

Reply:
I have had trouble with certain old versions of VCS, using bool(s) caused me problem. I suggest to keep the refactoring to coding styles. There are some portions of these structs that map to SV-DPI, and they can be particularly tricky and may render the code inoperable. It is extremely difficult to narrow down problems to particular versions of Modelsim and VCS.

Reply:
OK, will keep the "#define ASE_BUFFER_VALID        0xFFFF ". Thank you for your feedbacks.